arch/posix/include/kernel_arch_thread.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2017 Intel Corporation
+ * Copyright (c) 2017 Oticon A/S
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Per-arch thread definition
+ *
+ * This file contains definitions for
+ *
+ *  struct _thread_arch
+ *  struct _callee_saved
+ *  struct _caller_saved
+ *
+ * necessary to instantiate instances of struct k_thread.
+ */
+
+#ifndef _kernel_arch_thread__h_
+#define _kernel_arch_thread__h_
+
+#ifndef _ASMLANGUAGE
+#include <zephyr/types.h>
+
+struct _caller_saved {
+	/*
+	 * Nothing here
+	 */
+};
+
+typedef struct _caller_saved _caller_saved_t;
```

Comment:
In general for all those new types you've created, the preferred approach is to use `struct` instead and remove the `_t`

Reply:
This is just like in all other archs. Those structures are part of the kernel-arch IF.